File: readthedocs/code-tabs/java/src/main/java/nl/esciencecenter/xenon/tutorial/SlurmJobListGetter.java
package nl.esciencecenter.xenon.tutorial;

import java.util.Map;

import nl.esciencecenter.xenon.XenonException;
import nl.esciencecenter.xenon.credentials.PasswordCredential;
import nl.esciencecenter.xenon.schedulers.Scheduler;

public class SlurmJobListGetter {

    public static void main(String[] args) throws XenonException {

        String host = "localhost";
        String port = "10022";
        Map<String, String> properties = null;

        runExample(host, port, properties);
    }

    public static void runExample(String host, String port, Map<String, String> properties) throws XenonException {


        String adaptor = "slurm";
        String location = "ssh://" + host + ":" + port;

        // make the password credential for user 'xenon'
        String username = "xenon";
        char[] password = "javagat".toCharArray();
        PasswordCredential credential = new PasswordCredential(username, password);

        // create the SLURM scheduler
        Scheduler scheduler = Scheduler.create(adaptor, location, credential, properties);

        // ask SLURM for its queues and list the jobs in each
        for (String queueName : scheduler.getQueueNames()) {
            System.out.println("List of jobs in queue '" + queueName + "' for SLURM at " + location + "");
            String[] jobs = scheduler.getJobs(queueName);
            if (jobs.length == 0) {
                System.out.println("(No jobs)");
            } else {
                for (String job : jobs) {
                    System.out.println(job);
                }
            }
        }
    }
}

File: readthedocs/code-tabs/python/pyxenon_snippets/slurm_job_list_getter.py
import xenon
from xenon import PasswordCredential, Scheduler


def run_example():

    xenon.init()

    credential = PasswordCredential(username='xenon',
                                    password='javagat')

    location = 'ssh://localhost:10022'

    scheduler = Scheduler.create(adaptor='slurm',
                                 location=location,
                                 password_credential=credential)

    for queue_name in scheduler.get_queue_names():
        print("List of jobs in queue {queue_name} for SLURM at {location}"
              .format(queue_name=queue_name, location=location))
        jobs = scheduler.get_jobs([queue_name])
        if not jobs:
            print("(No jobs)")
        else:
            for job in jobs:
                print("    {}".format(job))

    scheduler.close()


if __name__ == '__main__':
    run_example()
